Network/Haskoin/Crypto/Keys.hs

```diff
@@ -83,6 +85,27 @@ data PubKeyI c = PubKeyI
 instance NFData (PubKeyI c) where
     rnf (PubKeyI p c) = rnf p `seq` rnf c
 
+instance ToJSON (PubKeyI Generic) where
+    toJSON = String . pack . bsToString . encode'
```

Comment:
This does not yield a hex-encoded string, which is the way we expect this kind of data to be encoded in JSON. It is better to convert to hex with bsToHex.